server/src/main/java/timely/server/store/MetricAgeOffIterator.java

```diff
@@ -127,16 +128,34 @@ private void handleNewMetricName(String metricName) {
         }
     }
 
-    private void seekPastAgedOffMetricData(String metricName, long ageOffTime) {
+    /**
+     * Ensure that we have a row that is following that of the current top row. If the newStart is not after the topRow, then return a row that is just after
+     * topRow.
+     *
+     * @param topRow
+     * @param newStart
+     * @return the new start
+     */
+    private byte[] ensureFollowing(byte[] topRow, byte[] newStart) {
```

Comment:
My only suggestion would be to add a log statement so we could see what the values are when the next row is not after the top row to better troubleshoot.